### PR TITLE
Support mode attribute on GCS files

### PR DESCRIPTION
**Pull request by charlesccychen**

### CODE REVIEW

**Comment by charlesccychen** [comment]
R: @aaltay

**Comment by aaltay** [comment]
LGTM

**Comment by charlesccychen** [comment]
Thanks! R: @dhalperi

File: sdks/python/apache_beam/io/gcsio.py [hunk 1]
```diff
@@ -111,10 +111,11 @@ def open(self, filename, mode='r',
       ValueError: Invalid open file mode.
     """
     if mode == 'r' or mode == 'rb':
-      return GcsBufferedReader(self.client, filename,
+      return GcsBufferedReader(self.client, filename, mode=mode,
```
Comment (dhalperi) [inline sdks/python/apache_beam/io/gcsio.py]: what does `mode='w'` on a reader mean?

File: sdks/python/apache_beam/io/gcsio_test.py [hunk 2]
```diff
@@ -218,6 +218,13 @@ def test_size(self):
     self.assertTrue(self.gcs.exists(file_name))
     self.assertEqual(1234, self.gcs.size(file_name))
 
+  def test_file_mode(self):
+    file_name = 'gs://gcsio-test/dummy_mode_file'
+    with self.gcs.open(file_name, 'wb') as f:
+      assert f.mode == 'wb'
+    with self.gcs.open(file_name, 'rb') as f:
+      assert f.mode == 'rb'
+
```
Comment (dhalperi) [inline sdks/python/apache_beam/io/gcsio_test.py]: maybe a test that it rejects the invalid modes like `r+b` and `w+`?

Comment (charlesccychen) [inline sdks/python/apache_beam/io/gcsio_test.py]: Done.

**Comment by charlesccychen** [comment]
Thanks, PTAL.

**Comment by dhalperi** [comment]
LGTM, merged, close at leisure :)

**Comment by charlesccychen** [comment]
Thanks Dan!

**Pull request closed without merging**